pkg/3scale/amp/operator/system_options_provider.go

```diff
@@ -433,7 +433,9 @@ func (s *SystemOptionsProvider) setFileStorageOptions() {
 		s.apimanager.Spec.System.FileStorageSpec.S3 != nil {
 		s.options.S3FileStorageOptions = &component.S3FileStorageOptions{
 			ConfigurationSecretName: s.apimanager.Spec.System.FileStorageSpec.S3.ConfigurationSecretRef.Name,
+			S3STSEnabled:            s.apimanager.Spec.System.FileStorageSpec.S3.S3STSEnabled,
 		}
+		s.options.S3FileStorageOptions.S3STSEnabled, _ = s.isSTS()
```

Comment:
unhandled error here

Reply:
Thank you for comment. I can remove error return, and write just notification.  As func (s *SystemOptionsProvider) setFileStorageOptions() {   is SET and does not return anything. What do you think? Or leave it as is?

Reply:
IMO `setFileStorageOptions` method now should return an error. The only source of error can be that the secret cannot be found, and from this context, it should exists.  So, the operator should return an error and retry.

Reply:
done. Thank you